homeassistant/components/light/mqtt.py

```diff
@@ -192,6 +192,10 @@ def turn_on(self, **kwargs):
         """Turn the device on."""
         should_update = False
 
+        if not self._state:
```

Comment:
Exactly. There is no need for me to have a state topic as the ESP accept only a pwm value to a pin OR a on/off payload. Once you sent a pwm value, it does not accept a On/off payload anymore. So for my case the state topic is useless unless i only want to use on off function without dimming